applications/GeoMechanicsApplication/tests/cpp_tests/test_apply_c_phi_reduction_process.cpp

```diff
@@ -0,0 +1,169 @@
+    Vector umat_parameters(6);
+    umat_parameters <<= 10000000, 0.2, 10.0, 25.0, 25.0, 1000;
+    r_model_part_properties.SetValue(UMAT_PARAMETERS, umat_parameters);
+    r_model_part_properties.SetValue(INDEX_OF_UMAT_C_PARAMETER, 3);
+    r_model_part_properties.SetValue(INDEX_OF_UMAT_PHI_PARAMETER, 4);
+    r_model_part_properties.SetValue(NUMBER_OF_UMAT_PARAMETERS, 6);
+
+    return result;
+}
+
+void CheckReducedCPhi(const ModelPart& rModelPart, double COrig, double PhiOrig, double ReductionFactor)
+{
+    for (Element& rElement : rModelPart.Elements()) {
+        const auto element_properties     = rElement.GetProperties();
+        const auto umat_properties_vector = element_properties.GetValue(UMAT_PARAMETERS);
+        const auto c_index   = element_properties.GetValue(INDEX_OF_UMAT_C_PARAMETER) - 1;
+        const auto phi_index = element_properties.GetValue(INDEX_OF_UMAT_PHI_PARAMETER) - 1;
+
+        KRATOS_EXPECT_DOUBLE_EQ(umat_properties_vector(c_index), ReductionFactor * COrig);
+
+        const double phi_rad = MathUtils<>::DegreesToRadians(PhiOrig);
+        const double tan_phi = std::tan(phi_rad);
+        KRATOS_EXPECT_DOUBLE_EQ(std::tan(MathUtils<>::DegreesToRadians(umat_properties_vector(phi_index))),
+                                ReductionFactor * tan_phi);
+    };
+}
+
+void CheckThrow(ModelPart& rModelPart)
+{
+    auto has_thrown = false;
+
+    try {
+        ApplyCPhiReductionProcess process{rModelPart, {}};
+        process.ExecuteInitializeSolutionStep();
+    } catch (const Exception&) {
+        has_thrown = true;
+    }
+
+    KRATOS_EXPECT_TRUE(has_thrown)
+}
+} // namespace
+
+namespace Kratos::Testing
+{
+KRATOS_TEST_CASE_IN_SUITE(CheckCAndPhiReducedAfterCallingApplyCPhiReductionProcess, KratosGeoMechanicsFastSuite)
+{
+    Model model;
+    auto& r_model_part = PrepareCPhiTestModelPart(model);
+
+    ApplyCPhiReductionProcess process{r_model_part, {}};
+    process.ExecuteInitializeSolutionStep();
+
+    CheckReducedCPhi(r_model_part, 10.0, 25.0, 0.9);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCAndPhiTwiceReducedAfterCallingApplyCPhiReductionProcessTwice, KratosGeoMechanicsFastSuite)
+{
+    Model model;
+    auto& r_model_part = PrepareCPhiTestModelPart(model);
+
+    ApplyCPhiReductionProcess process{r_model_part, {}};
+    process.ExecuteInitializeSolutionStep();
+    process.ExecuteFinalizeSolutionStep();
+    process.ExecuteInitializeSolutionStep();
+
+    CheckReducedCPhi(r_model_part, 10.0, 25.0, 0.8);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckFailureUmatInputsApplyCPhiReductionProcess, KratosGeoMechanicsFastSuite)
+{
+    Model model;
+    auto& r_model_part = SetGeometryAndMesh(model);
+
+    auto& r_model_part_properties = r_model_part.GetProperties(0);
+    auto  p_dummy_law             = std::make_shared<Testing::StubLinearElasticLaw>();
+    r_model_part_properties.SetValue(CONSTITUTIVE_LAW, p_dummy_law);
+
+    // check absence of UMAT_PARAMETERS
+    CheckThrow(r_model_part);
```

Comment:
I would suggest to use `KRATOS_EXPECT_EXCEPTION_IS_THROWN` rather than your custom check function. The reason is that the Kratos macro also checks the error message (i.e. whether it correctly sees that `UMAT_PARAMETERS` is missing) rather than accepting _any_ exception. The same suggestion applies to the next checks for throwing exceptions.

Reply:
Done